src/condor_utils/proc_family_direct_cgroup_v2.cpp

```diff
@@ -426,61 +513,38 @@ ProcFamilyDirectCgroupV2::kill_family(pid_t pid)
 	// Suspend the whole cgroup first, so that all processes are atomically
 	// killed
 	this->suspend_family(pid);
-	this->signal_process(pid, SIGKILL);
-	this->continue_family(pid);
-	return true;
-}
 
-// Return a vector of the absolute paths to all the
-// cgroup directories rooted here.
-static std::vector<stdfs::path> getTree(std::string cgroup_name) {
-	std::vector<stdfs::path> dirs;
-
-	// First add the root of the tree, the iterator below skips it
-	dirs.emplace_back(cgroup_mount_point() / cgroup_name);
-
-	// append all directories from here on down
-	for (auto entry: stdfs::recursive_directory_iterator{cgroup_mount_point() / cgroup_name}) {
-		if (stdfs::is_directory(entry)) {
-			dirs.emplace_back(entry);
-		}	
+	// This might not exist in your implementation of cgroupv2
+	stdfs::path kill_path = cgroup_mount_point() / stdfs::path(cgroup_name) / stdfs::path("cgroup.kill");
+	FILE *f = fopen(kill_path.c_str(), "r");
+	if (f) {
+		fprintf(f, "%c", '1');
+		fclose(f);
```

Comment:
There's a big code clone here with trimCgroupTree(). We should move the "kill everything" bits into their own function.